src/components/Tabs/Tabs.tsx

```diff
@@ -1,35 +1,32 @@
 import { FC } from 'react';
 import { Tab } from '../../types/tab';
+import classNames from 'classnames';
 
 type Props = {
   tabs: Tab[];
   selectedTabId: string;
-  onTabSelected: (tab: Tab) => void;
+  onTabSelected: (clicked: Tab) => void;
 };
 
 export const Tabs: FC<Props> = ({ tabs, selectedTabId, onTabSelected }) => {
   const selected = tabs.find(tab => tab.id === selectedTabId) || tabs[0];
 
-  const handleClick = (tab: Tab) => {
-    if (selected.id !== tab.id) {
-      onTabSelected(tab);
-    }
-  };
-
   return (
     <div data-cy="TabsComponent">
       <div className="tabs is-boxed">
         <ul>
           {tabs.map(tab => (
             <li
-              className={selected.id === tab.id ? 'is-active' : ''}
+              className={classNames({
+                'is-active': selected?.id === tab.id,
+              })}
               data-cy="Tab"
               key={tab.id}
             >
               <a
                 href={`#${tab.id}`}
                 data-cy="TabLink"
-                onClick={() => handleClick(tab)}
+                onClick={() => onTabSelected(tab)}
```

Comment:
> After fixes test "should not invoke onTabSelected if the current tab is clicked" not passing, even when worked like need to be

it's because now at onClick you have onTabSelected instead of handleClick ;p change back to handleClick and condition to invoke onTabSelected also should be there again